### ISSUE TITLE
[gcc] Various atomics build errors on GCC 9

### ISSUE THREAD

**Issue by stellaraccident** [google/iree#4372]
Trying to enable a CI for this config and hitting errors.

```
/project/main_checkout/iree/base/synchronization.c: In function ‘iree_slim_mutex_deinitialize’:
/project/main_checkout/iree/base/synchronization.c:379:7: error: expected ‘)’ before ‘__extension__’
  379 |       iree_atomic_load_int32(&mutex->value, iree_memory_order_seq_cst) == 0);
      |       ^~~~~~~~~~~~~~~~~~~~~~
/project/main_checkout/iree/base/synchronization.c:42:31: note: in definition of macro ‘SYNC_ASSERT’
   42 | #define SYNC_ASSERT(x) (void)(x)                                                                                                                                         |                               ^
/project/main_checkout/iree/base/synchronization.c:42:30: note: to match this ‘(’
   42 | #define SYNC_ASSERT(x) (void)(x)                                                                                                                                         |                              ^
/project/main_checkout/iree/base/synchronization.c:378:3: note: in expansion of macro ‘SYNC_ASSERT’
  378 |   SYNC_ASSERT(
      |   ^~~~~~~~~~~
```

**Comment by stellaraccident** [comment]
Simple fix to the iree_atomic_load_auto macro, which seems to have gotten some copy-pasta stuck in it.

```
#define iree_atomic_load_auto(object, order)                       \
  __extension__({                                                  \
    __iree_auto_type __atomic_load_ptr = (object);                 \
    __typeof__(*__atomic_load_ptr) __atomic_load_tmp;              \
    __atomic_load(__atomic_load_ptr, &__atomic_load_tmp, (order)); \
    __atomic_load_tmp;                                             \
  })
```

**Comment by stellaraccident** [comment]
Fixed with commit 723ff372291fe035d197bf1acd56c59cca14c94f
